AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/MutationRetryNotifier.swift

```diff
@@ -69,6 +82,10 @@ extension MutationRetryNotifier: Subscriber {
 
     func receive(completion: Subscribers.Completion<Never>) {
         log.verbose(#function)
+        lock.lock()
```

Comment:
Suggestion. Instead of having this same code everywhere you could write a helper function in this class for this lock like:
```
    private func withLockingClosure<T>(closure: () -> T) -> T {
        lock.lock()
        defer {
            lock.unlock()
        }
        return closure()
    }
```

Reply:
Good pattern, thanks for the suggestion!

Reply:
In fact, this is broadly applicable across Amplify. I'll put it in a separate PR

Reply:
sorry I am just randomly stalking your PRs heh, saw this in my email and I had literally written the same code as you not a day before and it bothered me to write lock.lock() and lock.unlock() like 5 times in the same file so figured I could pass the result of my frustration along!